modules/nf-core/atlas/splitmerge/meta.yml

```diff
@@ -7,58 +7,71 @@ keywords:
   - read group
 tools:
   - "atlas":
-      description: "ATLAS, a suite of methods to accurately genotype and estimate genetic diversity"
+      description: "ATLAS, a suite of methods to accurately genotype and estimate genetic
+        diversity"
       homepage: "https://bitbucket.org/wegmannlab/atlas/wiki/Home"
       documentation: "https://bitbucket.org/wegmannlab/atlas/wiki/Home"
       tool_dev_url: "https://bitbucket.org/wegmannlab/atlas"
       doi: "10.1101/105346"
-      licence: "['GPL v3']"
+      licence: ["GPL v3"]
+      identifier: biotools:atlas_db
 input:
-  - meta:
-      type: map
-      description: |
-        Groovy Map containing sample information
-        e.g. [ id:'test', single_end:false ]
-  - bam:
-      type: file
-      description: Single input BAM file.
-      pattern: "*.bam"
-  - bai:
-      type: file
-      description: The BAI file for the input BAM file
-      pattern: "*.bai"
-  - read_group_setting:
-      type: file
-      description: |
-        TXT file containing the split and merge settings for
-        each readgroup. Each line consist of one readgroup,
-        single/double identifier and the maximum cycle number
-        of the sequencer. e.g. "RG1 single 100"
-      pattern: "*.txt"
-  - blacklist:
-      type: file
-      description: |
-        blacklist.txt (optional), A txt file with blacklisted read names
-        that should be ignored and just written to file, each on a new line
-      pattern: "*.txt"
+  - - meta:
+        qualifier: val
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. [ id:'test', single_end:false ]
+    - bam:
+        qualifier: path
+        type: file
+        description: Single input BAM file.
+        pattern: "*.bam"
+    - bai:
+        qualifier: path
+        type: file
+        description: The BAI file for the input BAM file
+        pattern: "*.bai"
+    - read_group_settings:
+        type: file
+        description: |
+          TXT file containing the split and merge settings for
+          each readgroup. Each line consist of one readgroup,
+          single/double identifier and the maximum cycle number
+          of the sequencer. e.g. "RG1 single 100"
+        pattern: "*.txt"
+        qualifier: path
+    - blacklist:
+        qualifier: path
+        type: file
+        description: |
+          blacklist.txt (optional), A txt file with blacklisted read names
+          that should be ignored and just written to file, each on a new line
+        pattern: "*.txt"
 output:
-  - meta:
-      type: map
-      description: |
-        Groovy Map containing sample information
-        e.g. [ id:'test', single_end:false ]
+  - data:
+      - meta:
+          qualifier: val
+          type: map
+          description: |
+            Groovy Map containing sample information
+            e.g. [ id:'test', single_end:false ]
+      - "*_mergedReads.bam":
```

Comment:
The output channels seem to have lost their names.

Reply:
The name of the output channel is `data` in [this case](https://github.com/nf-core/modules/blob/master/modules/nf-core/atlas/splitmerge/main.nf#L14). And we specify each element with an array, here we have `meta` and two files. To make it as automated as possible, we are copying what's written in the `main.nf` as output, which is the pattern to the files.
I agree it's not the prettiest, specially in some cases where this regex can get more complicated, but I think it's the best way to do it, instead of having custom names which can't be read automatically from the `main.nf` file.

Reply:
Ok, I see